src/parser/nmodl_driver.cpp

```diff
@@ -21,12 +21,18 @@ NmodlDriver::NmodlDriver(bool strace, bool ptrace)
 
 /// parse nmodl file provided as istream
 std::shared_ptr<ast::Program> NmodlDriver::parse_stream(std::istream& in) {
-    NmodlLexer scanner(*this, &in);
-    NmodlParser parser(scanner, *this);
+    scanner = new NmodlLexer(*this, &in);
+    parser = new NmodlParser(*scanner, *this);
```

Comment:
could these be smart pointers?

Reply:
Probably - and you're right, they should. Unfortunately converting to `unique_ptr` causes some test cases to segfault again...

Reply:
This was a very bad idea altogether! I have a better implementation now.

Reply:
👌  (I must admit I didn't look at the code locally while reviewing PR. Just looked at the diff and thought reasonable!)